crates/rome_fs/src/fs/os.rs

```diff
@@ -338,6 +286,72 @@ fn handle_dir_entry<'scope>(
     }));
 }
 
+/// Handles symlinks by recursively following them (up to `MAX_SYMLINK_DEPTH`),
+/// and returning the new type of the file pointed to. Performs error reporting,
+/// so the returned error type is `()`.
```

Comment:
```suggestion
/// Handles symlinks by recursively following them up to [MAX_SYMLINK_DEPTH],
/// and returning the type of file pointed to. Performs error reporting,
/// so the returned error type is `()`.
```

Could you add a doc section like this:

```markdown
///  ## Returns
/// 
///  Returns a tuple where the first argument is ... and the second is ...
```